app/controllers/service_specific_information_controller.rb

```diff
@@ -1,11 +1,8 @@
 class ServiceSpecificInformationController < ApplicationController
-  def index
-    grouped_posts = ServiceSpecificPost.all.group_by(&:branch)
-
-    @army_posts = grouped_posts['army']
-    @air_force_posts = grouped_posts['air_force']
-    @navy_posts = grouped_posts['navy']
-    @marine_corps_posts = grouped_posts['marine_corps']
-    @coast_guard_posts = grouped_posts['coast_guard']
+  def show
+    branch_id = params[:id] || 'army'
+    @branch = BranchOfService.includes(:service_specific_posts, :branch_of_service_contact).find(branch_id)
+    @branches = BranchOfService.all
+    render branch_id
```

Comment:
I'm getting a security warning for this: 

Confidence: High
Category: Dynamic Render Path
Check: Render
Message: Render path contains parameter value
Code: render(action => (params[:id] or "army"), {})

I'm not sure if there's any way to avoid this in the currently way this is set up aside from putting something like the switch statement back.